protobuf/sampling.proto

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package jaegertracing.protobuf.sampling_manager;
+
+import "jaeger.proto";
+
+message ProbabilisticSamplingStrategy {
+    double sampling_rate = 1;
+}
+
+message RateLimitingSamplingStrategy {
+    int32 max_traces_per_seconds = 1;
+}
+
+message OperationSamplingStrategy {
+    string operation = 1;
+    oneof strategy {
+        ProbabilisticSamplingStrategy probabilistic = 2;
+        RateLimitingSamplingStrategy rate_limiting = 3;
+    }
+}
+
+message PerOperationSamplingStrategy {
+    double default_sampling_probability = 1;
+    double default_lower_bound_traces_per_second = 2;
+    repeated OperationSamplingStrategy per_operation_strategy = 3;
+    double default_upper_bound_traces_per_second = 4;
```

Comment:
nit: group with other primitive fields

Reply:
Actually this is one of the more interesting issues with Protobuf v3. I think in Thrift, the `defaultUpperBoundTracesPerSecond` field is the only one marked as optional. Protobuf v3 makes all fields optional, so it makes sense. This default optional behavior may affect how we handle zero values.